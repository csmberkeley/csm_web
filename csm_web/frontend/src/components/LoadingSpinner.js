import React from "react";
import PropTypes from "prop-types";

export default function LoadingSpinner({ id }) {
  return (
    <div className="sk-fading-circle" id={id}>
      {[...Array(12)].map((_, i) => (
        <div key={i} className={`sk-circle${i + 1} sk-circle`} />
      ))}
    </div>
  );
}

LoadingSpinner.propTypes = { id: PropTypes.string };
