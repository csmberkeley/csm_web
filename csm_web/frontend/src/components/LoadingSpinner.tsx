import React from "react";

interface LoadingSpinnerProps {
  id: string;
}

export default function LoadingSpinner({ id }: LoadingSpinnerProps): JSX.Element {
  return (
    <div className="sk-fading-circle" id={id}>
      {[...Array(12)].map((_, i) => (
        <div key={i} className={`sk-circle${i + 1} sk-circle`} />
      ))}
    </div>
  );
}
