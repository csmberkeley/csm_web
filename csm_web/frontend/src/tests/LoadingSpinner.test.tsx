import React from "react";
import { render } from "@testing-library/react";
import LoadingSpinner from "../components/LoadingSpinner";

describe("LoadingSpinner", () => {
  it("should render correctly", () => {
    const component = render(<LoadingSpinner />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
