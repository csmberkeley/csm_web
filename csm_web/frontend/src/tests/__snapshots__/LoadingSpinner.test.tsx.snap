// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sk-fading-circle "
  >
    <div
      class="sk-circle1 sk-circle"
    />
    <div
      class="sk-circle2 sk-circle"
    />
    <div
      class="sk-circle3 sk-circle"
    />
    <div
      class="sk-circle4 sk-circle"
    />
    <div
      class="sk-circle5 sk-circle"
    />
    <div
      class="sk-circle6 sk-circle"
    />
    <div
      class="sk-circle7 sk-circle"
    />
    <div
      class="sk-circle8 sk-circle"
    />
    <div
      class="sk-circle9 sk-circle"
    />
    <div
      class="sk-circle10 sk-circle"
    />
    <div
      class="sk-circle11 sk-circle"
    />
    <div
      class="sk-circle12 sk-circle"
    />
  </div>
</DocumentFragment>
`;
